### PR TITLE
Adds ability to set template context values on command line 

### PR DESCRIPTION
**Pull request by gwkunze**
Following up from #4668, Adapted from helm (a couple lines copied/altered) allows you to specify template context values on the command line.

For example:
```
$ kops toolbox template --set foo=bar,baz=3 --set-string ban=4 mytemplate.yml
```

Would be equivalent to:
```
$ kops toolbox template --values values.yml mytemplate.yml
```

if `values.yml` contains:
```
---
foo: bar
baz: 3
ban: "4"
```

### CODE REVIEW

**Comment by gwkunze** [comment]
I haven't really dived in to the whole bazel thing, but I don't really understand why a simple `dep ensure -add k8s.io/helm/pkg/strvals` would replace all the `BUILD.bazel` files with different `BUILD` files, is there maybe some command I need to run besides `./hack/update-bazel.sh`?

**Comment by mikesplain** [comment]
/ok-to-test

@gwkunze Looks like you might need to run

`make dep-ensure` or at least run the find commands to strip them out:
https://github.com/kubernetes/kops/blob/f703888b417a0e4409d2697fd56169a57ac3b7f7/Makefile#L451-L455

**Comment by gwkunze** [comment]
@mikesplain thanks for the info.

**Comment by mikesplain** [comment]
@gwkunze Thanks for the contribution! This is great! Maybe in a follow up PR you could add some more docs to https://github.com/kubernetes/kops/blob/3f2e53778976317247ec29a90ab5a3bd34ec296c/docs/cluster_template.md 

/lgtm

**Comment by mikesplain** [comment]
/lgtm cancel

You bet! Go ahead we’ll remove the lgtm for now. Good discovery.

**Comment by gwkunze** [comment]
For now I didn't rewrite history to make it easier to review the new changes, if preferred I can squash the final two commits into the first one.

**Comment by mikesplain** [comment]
Very nice! Good fix and docs! Thanks!
/lgtm

File: cmd/kops/toolbox_template.go [hunk 1]
```diff
@@ -227,6 +233,20 @@ func newTemplateContext(files []string) (map[string]interface{}, error) {
 		}
 	}
 
+	// User specified a value via --set
+	for _, value := range values {
+		if err := strvals.ParseInto(value, context); err != nil {
+			return nil, fmt.Errorf("failed parsing --set data: %s", err)
```
Comment (chrislovecnm) [inline cmd/kops/toolbox_template.go]: should this be `%v`?

Comment (gwkunze) [reply cmd/kops/toolbox_template.go]: It shouldn't matter, according to golang's documentation on the [fmt package](https://golang.org/pkg/fmt/):
> If the format (which is implicitly %v for Println etc.) is valid for a string (%s %q %v %x %X), the following two rules apply:
> 4. If an operand implements the error interface, the Error method will be invoked to convert the object to a string, which will then be formatted as required by the verb (if any).

**Comment by gwkunze** [comment]
I've attempted to rebase but suddenly I'm getting a lot of extra libraries installed and running `hack/update-bazel.sh` yields `gazelle: -go_prefix not set, and no # gazelle:prefix directive found in go/src/k8s.io/kops/BUILD.bazel` which seems weird to me. I've pushed the resulting code to https://github.com/usabilla/kops/commit/c5b04436c4b6d5c4924ee75a7bdf2d6c63de7f36 but since I'm not sure how to fix this I'll not push it to this PR yet. Any help would be appreciated.

**Comment by mikesplain** [comment]
@gwkunze Hmm, I attempted your rebase here (https://github.com/mikesplain/kops/tree/set-values) but I don't think I ended up with some of the binaries you did.  

My history:
```
git checkout https://github.com/kubernetes/kops/pull/5108
git rebase origin/master
# resolve merges, newer version of helm, keep both lines in BUILD.bazel
git add .
git rebase --continue
```

Once complete, running `make bazel-gazelle` shows everythings up to date. Hope this helps

**Comment by gwkunze** [comment]
@mikesplain thanks, I was reluctant to attempt to resolve the conflict in the `Gopkg.lock` file so I did a `git checkout --theirs` on it and ran `make dep-ensure` which caused all the extra packages to appear, I guess that installed a newer version of a dep which cause my problems. It should be correct now.

**Comment by justinsb** [comment]
Thanks @gwkunze - I don't love that we're vendoring helm, but as we're only including a small piece of it, I guess it is OK :-)

/approve
/lgtm

**Pull request merged**